### PR TITLE
Fix cookie implementation and use specific cookie for lang

### PR DESCRIPTION
**Pull request by sebastienbeau**
Fix the current implementation for getting and setting the cookie. The
set method was not working. Add a new steam variable 'steam.cookies'
that can be extended in your custom gem to add aditionnal cookies
Change the way to store the selected lang. Instead of storing the
information in the main cookie, use a dedicated cookie.
This allow caching with external tools like varnish.
Adapt the test regarding this change
Add a docker configuration to run the test easily

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,39 @@ see the list in the issues section.
 4. Push to the branch (`git push origin my-new-feature`)
 5. Create new Pull Request
 
+## Running test with docker
+
+A docker configuration is included in this repository, you can run all the test with the following process
```
Comment (did) [inline README.md]: typo: ```run all the tests with the following procedure```

File: spec/unit/middlewares/locale_spec.rb [hunk 2]
```diff
@@ -9,29 +9,35 @@
...
   let(:url)             { 'http://models.example.com' }
   let(:app)             { ->(env) { [200, env, 'app'] } }
-  let(:services)        { instance_double('Services', :locale= => 'en') }
+  let(:cookie_lang)     { nil }
+  let(:cookie_service)  { instance_double('Cookie Service', :get => cookie_lang) }
+  let(:services)        { instance_double('Services', :locale= => 'en', :cookie => cookie_service) }
   let(:middleware)      { Locomotive::Steam::Middlewares::Locale.new(app) }
-  let(:session)         { {} }
   let(:accept_language) { '' }
+  let(:expected_lang)   { :de }
```
Comment (did) [inline spec/unit/middlewares/locale_spec.rb]: by convention, we don't use the `let` keyword to store expected results. It's better to read the expected result in the test itself.

File: spec/unit/services/cookie_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+require 'spec_helper'
+
+describe Locomotive::Steam::CookieService do
+
+  let(:steam_cookies)   { {} }
+  let(:request_cookies) { {} }
+  let(:request)         { instance_double('Request', env: { 'steam.cookies' => steam_cookies }, cookies: request_cookies) }
+  let(:cookie)          { {'value' => 'bar2'} }
+  let(:service)   { described_class.new(request)}
```
Comment (did) [inline spec/unit/services/cookie_service_spec.rb]: indentation :-)

File: spec/unit/services/action_service_spec.rb [hunk 4]
```diff
@@ -114,20 +115,19 @@
...
-      describe 'sendCookiesProp' do
-
-        let(:script) { "return setCookiesProp('done', true);" }
-
-        it { subject; expect(cookies['done']).to eq 'true' }
-
+      describe 'setCookiesProp' do
+        let(:script) { "return setCookiesProp('done', {'value': true});" }
+        before do
+          expect(cookie_service).to receive(:set).with('done', {'value' => true})
```
Comment (did) [inline spec/unit/services/action_service_spec.rb]: it'd better to move this statement within its own test

Comment (sebastienbeau) [reply spec/unit/services/action_service_spec.rb]: If you have some minute, I have a question on this

Comment (sebastienbeau) [reply spec/unit/services/action_service_spec.rb]: I think I got it, tell me if it's good.
I am newbie in ruby testing ;)

File: spec/unit/services/cookie_service_spec.rb (hunk 3 above)
Comment (sebastienbeau) [reply spec/unit/services/cookie_service_spec.rb]: what do you think about using rubocop + pre-commit with auto formating of the code and checking it localy before commit?
https://github.com/rubocop-hq/rubocop
https://pre-commit.com/
On python side we start to use same stuff (black that format all the code)

**Pull request merged**